Source/Scene/ShadowMapShader.js

```diff
@@ -17,6 +17,7 @@ define([
     };
 
     ShadowMapShader.createShadowCastVertexShader = function(vs, isPointLight, isTerrain) {
+        vs.defines.push('SHADOW_MAP');
```

Comment:
Move this to after `slice` so it doesn't modify the input source.

Reply:
It only works if I modify the input source, because the define needs to be added to the `ShaderSource` that's using the `vertexLogDepth` shader.  Is there a different way to do this?

Reply:
The log depth shaders don't use that varying. Only log depth + shadows use it.